internal/coreinternal/aggregateutil/aggregate.go

```diff
@@ -137,8 +137,21 @@ func isNotPresent(target string, arr []string) bool {
 	return true
 }
 
+func sortMapKeys[K any](input map[string]K) []string {
```

Comment:
This adds extra overhead during hotpath execution that is only needed for the test cases, as the actual order of the datapoints doesn't matter.  

Is there a way to update the test to not depend on the datapoint order or update the API to make sorting optional?

Reply:
Could something like `pmetrictest.CompareMetrics` help?

Reply:
Fair point, yes `compareMetrics` can help. but I think we need to implement a new `compareMetricsOption` to ignore the order of datapoints, since the comparison of the exact order happens [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/pdatatest/pmetrictest/metrics.go#L304)